materials/material.py

```diff
@@ -63,10 +65,63 @@ def __init__(self, name, form=None, condition=None, category=None, subcategory=N
         self.category = category
         self.subcategory = subcategory
         self.references = references
+        self.elemental_composition = elemental_composition
+        if self.elemental_composition is not None:
+            for element, limits in self.elemental_composition.items():
+                # limits[0] is element minimum percent by mass,
+                # limits[1] is element maximum percent by mass.
+                if not (0. <= limits[0] <= limits[1] <= 100.):
+                    raise ValueError(
+                        'Elemental composition limits on {:s} are not valid: '.format(element)
+                        + 'min = {:.3f} %, max = {:.3f} %'.format(*limits))
 
         if properties_dict is not None:
             self.properties = build_properties(properties_dict)
 
+    def __getitem__(self, key):
+        """Get a property of the material by name.
+        A Material is primarily a collection of properties, so we use []
+        as a shorthand to access a property.
+        i.e. `mat[key]` is a shorthand for `mat.properties[key]`.
+
+        Argument:
+            key (string): the name of a property.
+
+        Returns:
+            Property: `self.properties[key]`
+        """
+        if key not in self.properties:
+            raise KeyError(
+                'This Material does not have a {:s} property'.format(key)
+                + '\nThe valid keys are {}'.format(self.properties.keys())
+                )
+        return self.properties[key]
+
+    def elements_table_str(self):
+        """Create (as a string) a table of the elemental composition data."""
+        string = 'Elemental composition:\n'
+        sorted_by_max_mass = sorted(
+            self.elemental_composition.items(), key=lambda kv: kv[1][1], reverse=True)
```

Comment:
[taste] create a named variable that says what these indices are

`kv[VALUE_IND][WHATEVER_VAL_IND_1_IS_IND], `

Reply:
Fixed by https://github.com/mvernacc/material-properties-interchange/commit/b9add75bd7ed8e674f02c97e52227862e7ec1572